tests/test_primitives/test_memory.py

```diff
@@ -294,3 +294,40 @@ class test_memory_read_only(m.Circuit):
                            flags=["-Wno-unused"],
                            directory=os.path.join(os.path.dirname(__file__),
                                                   "build"))
+
+
+@pytest.mark.parametrize('RE', [True, False])
```

Comment:
if we're not doing functional testing, let's similarly move this to an MLIR test in examples.py